configurations/gdscript-configuration.json

```diff
@@ -1,36 +1,93 @@
 {
     "comments": {
         "lineComment": "#",
-        "blockComment": ["\"\"\"", "\"\"\""]
+        "blockComment": [
+            "\"\"\"",
+            "\"\"\""
+        ]
     },
     "brackets": [
-        ["(", ")"],
-        ["[", "]"],
-        ["{", "}"]
+        [
+            "(",
+            ")"
+        ],
+        [
+            "[",
+            "]"
+        ],
+        [
+            "{",
+            "}"
+        ]
     ],
     "autoClosingPairs": [
-        ["'", "'"],
-        ["\"", "\""],
-        ["(", ")"],
-        ["[", "]"],
-        ["{", "}"]
+        [
+            "'",
+            "'"
+        ],
+        [
+            "\"",
+            "\""
+        ],
+        [
+            "(",
+            ")"
+        ],
+        [
+            "[",
+            "]"
+        ],
+        [
+            "{",
+            "}"
+        ]
     ],
     "surroundingPairs": [
-        ["'", "'"],
-        ["\"", "\""],
-        ["(", ")"],
-        ["[", "]"],
-        ["{", "}"]
+        [
+            "'",
+            "'"
+        ],
+        [
+            "\"",
+            "\""
+        ],
+        [
+            "(",
+            ")"
+        ],
+        [
+            "[",
+            "]"
+        ],
+        [
+            "{",
+            "}"
+        ]
     ],
     "indentationRules": {
         "increaseIndentPattern": "^\\s*((class|static func|func|else|elif|for|if|match|while|enum)|(.*\\sdo\\b))\\b[^\\{;]*$",
         "decreaseIndentPattern": "^\\s*([}\\]]([,)]?\\s*(#|$)|\\.[a-zA-Z_]\\w*\\b)|(else|elif)\\b)"
     },
     "folding": {
-		"offSide": true,
-		"markers": {
-			"start": "^\\s*#\\s*region\\b",
-			"end": "^\\s*#\\s*endregion\\b"
-		}
-	}
+        "offSide": true,
+        "markers": {
+            "start": "^\\s*#\\s*region\\b",
+            "end": "^\\s*#\\s*endregion\\b"
+        }
+    },
+    "onEnterRules": [
+        {
+            "beforeText": "^\\s*(?:func|for|if|elif|else|while|match).*?:\\s*$",
```

Comment:
Is this also needed for `class`, `static func` and `enum` (as done above)?

Reply:
What I noticed when looking into your question, is that I don't need to override the behavior described in the `indentationRules` (more specifically, I think that the `indentationRules` takes priority over `onEnterRules`). Basically, only the second rule is needed for the #337 to be solved.